### PR TITLE
add server_info ffi for TiFlash

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Support `fn_server_info` to forward server info from proxy to TiFlash.

### What is changed and how it works?

What's Changed:

add `fn_server_info` to `RaftStoreProxyFFIHelper`.
add `ffi_server_info` as implementation of `fn_server_info`.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/server/src/server.rs [hunk 1]
```diff
@@ -147,12 +150,14 @@ pub unsafe fn run_tikv(config: TiKvConfig) {
                 )),
             };
 
-            let proxy_helper = RaftStoreProxyFFIHelper::new(&proxy);
+            let mut proxy_helper = RaftStoreProxyFFIHelper::new(&proxy);
```
Comment (solotzg) [inline components/server/src/server.rs:153]: I think `proxy_helper` should be immutable. You can try like
```suggestion
            let proxy_helper = {
                let mut proxy_helper = RaftStoreProxyFFIHelper::new(&proxy);
                proxy_helper.fn_server_info = Some(ffi_server_info);
                proxy_helper
            };
```
or make another function to implement.

File: components/server/src/server.rs [hunk 2]
```diff
@@ -147,12 +150,14 @@ pub unsafe fn run_tikv(config: TiKvConfig) {
...
 
             info!("set raft-store proxy helper");
 
             get_engine_store_server_helper().handle_set_proxy(&proxy_helper);
 
+            proxy_helper.fn_server_info = Some(ffi_server_info);
```
Comment (solotzg) [inline components/server/src/server.rs:159]: Maybe it's unsafe to update `proxy_helper` after `handle_set_proxy`.

File: raftstore-proxy/ffi/src/RaftStoreProxyFFI/@version [hunk 3]
```diff
@@ -1,4 +1,4 @@
-//ef3c0a05cdfd04ac7f95eac4a3be74d2//501000//
+//0e0095b0e500d4b94f4ee8962689b8de//501008//
 #pragma once
 #include <cstdint>
-namespace DB { constexpr uint32_t RAFT_STORE_PROXY_VERSION = 501000; }
\ No newline at end of file
+namespace DB { constexpr uint32_t RAFT_STORE_PROXY_VERSION = 501008; }
```
Comment (solotzg) [inline raftstore-proxy/ffi/src/RaftStoreProxyFFI/@version:4]: You can run cmd like `./target/debug/gen-proxy-ffi --overwrite-version 501001` to set version manually.

**Review by solotzg — COMMENTED** [review]
It's recommended to run `cargo fmt` before committing. Rest lgtm

File: components/server/src/server.rs (hunk 2 above)
Comment (CalvinNeo) [reply components/server/src/server.rs:159]: I think so, when handle_set_proxy is called, TiFlash has every reason to call functions under RaftStoreProxyFFIHelper

File: components/server/src/util.rs [hunk 4]
```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use kvproto::diagnosticspb::{ServerInfoRequest, ServerInfoResponse, ServerInfoType};
```
Comment (solotzg) [inline components/server/src/util.rs:3]: You can use `cargo clippy` to optimize code structure.
```
warning: unused import: `std::sync::Mutex`
  --> components/server/src/util.rs:16:5
   |
16 | use std::sync::Mutex;
   |     ^^^^^^^^^^^^^^^^

warning: unused import: `FutureExt`
  --> components/server/src/util.rs:12:23
   |
12 | use futures::future::{FutureExt, TryFutureExt};
   |                       ^^^^^^^^^

warning: unused import: `TryFutureExt`
  --> components/server/src/util.rs:12:34
   |
12 | use futures::future::{FutureExt, TryFutureExt};
   |                                  ^^^^^^^^^^^^

warning: unused import: `futures::sink::SinkExt`
  --> components/server/src/util.rs:13:5
   |
13 | use futures::sink::SinkExt;
   |     ^^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `futures::stream::StreamExt`
  --> components/server/src/util.rs:14:5
   |
14 | use futures::stream::StreamExt;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unnecessary `unsafe` block
  --> components/server/src/util.rs:76:5
   |
76 |     unsafe {
   |     ^^^^^^ unnecessary `unsafe` block
   |
   = note: `#[warn(unused_unsafe)]` on by default
```

**Review by solotzg — APPROVED** [review]
lgtm

**Pull request merged**